spec/unleash/feature_toggle_spec.rb

```diff
@@ -20,16 +20,126 @@
-      enabled: true,
-      strategies: [],
-      variants: nil
+      JSON.parse('{
+          "name": "test",
+          "enabled": true,
+          "strategies": [],
+          "variants": []
+        }')
       ) }
 
-    it 'should return true' do
+    it 'should return true if enabled, and default is true' do
       context = Unleash::Context.new(user_id: 1)
       expect(feature_toggle.is_enabled?(context, true)).to be_truthy
     end
+
+    it 'should return true if enabled, and default is false' do
+      context = Unleash::Context.new(user_id: 1)
+      expect(feature_toggle.is_enabled?(context, false)).to be_truthy
+    end
+  end
+
+  describe 'FeatureToggle with empty strategies and disabled toggle' do
+    let(:feature_toggle) { Unleash::FeatureToggle.new(
+      JSON.parse('{
+          "name": "Test.userid",
+          "description": null,
+          "enabled": false,
+          "strategies": [],
+          "variants": [],
+          "createdAt": "2019-01-24T10:41:45.236Z"
+        }')
+      ) }
+
+    it 'should return false if disabled and default is false' do
+      context = Unleash::Context.new(user_id: 1)
+      expect(feature_toggle.is_enabled?(context, false)).to be_falsey
+    end
+
+    it 'should return true if disabled and default is true' do
+      context = Unleash::Context.new(user_id: 1)
+      expect(feature_toggle.is_enabled?(context, true)).to be_truthy
+    end
+  end
+
+  describe 'FeatureToggle with userId strategy and enabled toggle' do
+    let(:feature_toggle) { Unleash::FeatureToggle.new(
+      JSON.parse('{
+          "name": "Test.userid",
+          "description": null,
+          "enabled": true,
+          "strategies": [
+            {
+              "name": "userWithId",
+              "parameters": {
+                "userIds": "12345"
+              }
+            }
+          ],
+          "variants": [],
+          "createdAt": "2019-01-24T10:41:45.236Z"
+        }')
+      ) }
+
+    it 'should return true if enabled, user_id matched, and default is true' do
+      context = Unleash::Context.new(user_id: "12345")
+      expect(feature_toggle.is_enabled?(context, true)).to be_truthy
+    end
+
+    it 'should return true if enabled, user_id matched, and default is false' do
+      context = Unleash::Context.new(user_id: "12345")
+      expect(feature_toggle.is_enabled?(context, false)).to be_truthy
+    end
+
+    it 'should return false if enabled, user_id unmatched, and default is true' do
+      context = Unleash::Context.new(user_id: "54321")
+      expect(feature_toggle.is_enabled?(context, true)).to be_falsey
+    end
+
+    it 'should return false if enabled, user_id unmatched, and default is false' do
+      context = Unleash::Context.new(user_id: "54321")
+      expect(feature_toggle.is_enabled?(context, false)).to be_falsey
+    end
+  end
+
+  describe 'FeatureToggle with userId strategy and disabled toggle' do
+    let(:feature_toggle) { Unleash::FeatureToggle.new(
```

Comment:
And here. :)

Reply:
fix on commit 1ca04ac72fd6edcd507860350b28d74d2172bb9f